File: Linear_Search.c
/*
Time complexity of linear search algorithm -
Base Case - O(1)
Average Case - O(n)
Worst Case -O(n)*/
#include<stdio.h>
int main()
{
	int n,i,data,found=0;
	printf("Enter the  umber of elements in  the array: ");
	scanf("%d",&n);
	int a[n];
	for(i=0;i<n;i++)
	{
		printf("Enter the %d element: ",(i+1));
		scanf("%d",&a[i]);
	}
	printf("Enter the number to be searched: ");
	scanf("%d",&data);
	for(i=0;i<n;i++)
	{
		if(a[i]==data)
		{
			printf("%d is present at location %d.\n",data,i+1);
			found=1;
			break;
		}
	}
	if(found==0)
		{
			printf("%d isn't present in the array.\n",data);
		}
		return 0;
}

File: Linear_Search.py
from array import *
def search(arr, x):
    for i in range(len(arr)):

        if arr[i] == x:
            return i

    return -1
arr=array('i',[])
n=int(input("Enter the length of the array"))
for i in range(n):
    z=int(input("Enter the value"))
    arr.append(z)
x=int(input("Enter the number to be searched"))
# Function call
result = search(arr, x)



if result != -1:
	print(f"{x} Element is present at index", str(result+1))
else:
	print("Element is not present in array")
